lib/g_searcher/search_results/search_result_parser.ex

```diff
@@ -0,0 +1,83 @@
+defmodule GSearcher.SearchResults.SearchResultParser do
+  @top_ads "#tads a"
+  @bottom_ads "#bottomads #tadsb a"
+  @search_results "#search .g div div > a"
+  @total_count "#result-stats"
+  @url_link "href"
+
+  @google_url "https://www.google.com/"
+
+  def parse(all_html) do
+    with {:ok, html_tree} <- Floki.parse_document(all_html),
+         {:ok, top_advertiser_urls} <- fetch_top_advertiser_urls(html_tree),
+         {:ok, regular_advertiser_urls} <- fetch_regular_advertiser_urls(html_tree),
+         {:ok, search_result_urls} <- fetch_search_result_urls(html_tree),
+         {:ok, total_number_of_results} <- fetch_total_result_count(html_tree) do
```

Comment:
Suggestion:
As the steps here running as serial, but it could run as parallel because each step is separated, so how about we improve the implementation to make each step running parallel @bterone

Reply:
Haven't played around with parallelism too much, I used `Task` for doing this in 9b04d8ff77f9b8a15f9ec264f64dd42a9912aada, could you let me know what you think about the implementation? 🤔  @andyduong1920